### PR TITLE
Prevent user deleting themselves

### PR DESCRIPTION
**Pull request by colinbruce**
Update user handling to prevent deleting themselves
* hide the link from user if viewing own record
* route them back to the details page with an error if
they spoof the delete function

[finishes #103483020]

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 09cdf837b58b9b61b4ee6d43955f1fefd3d8570e
  * Your subject line is longer than 50 characters

Guidelines are available at https://github.com/ministryofjustice/fr-staffapp

---
This message was auto-generated by https://gitcop.com

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -2,9 +2,7 @@ class UsersController < ApplicationController
   respond_to :html
   before_action :authenticate_user!
   before_action :find_user, only: [:edit, :show, :update, :destroy]
```
Comment (dotemacs) [inline app/controllers/users_controller.rb]: I know that you didn't touch this line, but what do you think of this:

`before_action :find_user, except: [:index]`

?

File: app/views/users/show.html.slim [hunk 2]
```diff
@@ -33,10 +33,11 @@ h2 Staff details
   .row.pad-top-fifteen-px
     .columns.small-12
       = link_to "Change your password", edit_user_registration_path(@user)
--if current_user.elevated?
+-if current_user.elevated? && current_user != @user
```
Comment (dotemacs) [inline app/views/users/show.html.slim]: Here I'd put some sort of method that would describe the condition in an clear way. Something like:

`if user_has_the_permissions_and_not_themselves`

**Comment by dotemacs** [comment]
Discussed it offline, approved

**Pull request merged**